tensorflow/python/ops/variable_scope.py

```diff
@@ -690,6 +695,34 @@ def _get_single_variable(self,
     return v
 
 
+  # Initialize variable when no initializer provided
+  def _get_default_initializer(self, shape=None, dtype=dtypes.float32):
+    """Use this function to provide a default and corresponding initializer of dtype
+
+    Args:
+      dtype: Type of the new or existing variable.(defaults to `DT_FLOAT`).
+
+    Returns:
+      initializer and initializing_from_value. See documentation of get_variable above.
+
+    Raises:
+      TypeError: When giving unsupported dtype.
+    """
+    if dtype.is_floating:
+      initializer = init_ops.uniform_unit_scaling_initializer()
+      initializing_from_value = False
+    elif dtype.is_string:
+      initializer = init_ops.strings_initializer(shape=shape, dtype=dtype.base_dtype)
+      initializing_from_value=True
+    elif dtype.is_integer or dtype.is_unsigned or dtype.is_bool or dtype.is_complex:
```

Comment:
i'm not convinced that initializing complex with zeros is reasonable in general.  do you have some references showing neural networks that use complex parameters and suggest that initializing them with zeros is the right way to do it?

certainly having this be an undocumented feature is problematic; and since complex are complex floating point, it also seems very reasonable to initialize them like regular floating point.